linode/manager
# refactor: [M3-8203] - Refactor Region Select

## Description 📝

## Why ❓ 
- When I started working on addressing the Linode Create v2 side effect renders, I was unable to achieve what I wanted to easily with the current implementation. It wasn't easy to get the full `Region` object in the `onChange` handler, which I need because I want to reference the region capabilities in my`onChange`. I'm doing this PR first so that I can more easily get the raw `Region` value onChange, also allowing us to clean up the `RegionSelectOption` abstraction.

## Changes 🔁
- Makes RegionSelect and RegionMultiSelect use `Region` type instead of the `RegionSelectOption` abstraction
- Changes prop `handleSelection` to `onChange` to be more consistent with most selects
- Changes prop `selectedId` to `value` to be more consistent with most selects
- Changes prop `handleDiabledRegion` to `disabledRegions` to simplify disabling region options
- Removes custom `isClearable` prop in favor of MUI's `disableClearable` so we can retain the typesafety of `onChange` being `(region:Region | null) => void` when `disableClearable` is `false` and  `(region: Region) => void` when `disableClearable` is `true`

## Preview 📷

> [!note]
> No UI changes expected

## How to test 🧪

- Test the Region Select throughout the application
- Test the RegionMultiSelect, which can be found on the Object Storage Access Keys page with **Mutli-Cluster** turned on
- Make sure Account Availability is working as expected
- Verify regions are shown in the correct order (should be the same order as production)
- Verify Distributed Compute Region icons render when expected

## As an Author I have considered 🤔
